src/averageAges.js

```diff
@@ -37,7 +50,12 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const ageOfWomen = people
+    .filter(({ name, sex }) => sex === FEMALE_SEX
+      && (withChildren ? people.some(({ mother }) => mother === name) : true))
+    .map(({ born, died }) => died - born);
```

Comment:
Now everything is written correctly, but it is so difficult to read and expand. Break it into smaller parts.